lib/classes/Variables.js

```diff
@@ -962,30 +962,65 @@ class Variables {
     );
   }
 
-  warnIfNotFound(variableString, valueToPopulate) {
+  _describeVariableType(variableString) {
+    if (variableString.match(this.envRefSyntax)) {
+      return 'environment variable';
+    } else if (variableString.match(this.optRefSyntax)) {
+      return 'option';
+    } else if (variableString.match(this.selfRefSyntax)) {
+      return 'service attribute';
+    } else if (variableString.match(this.fileRefSyntax)) {
+      return 'file';
+    } else if (variableString.match(this.ssmRefSyntax)) {
+      return 'SSM parameter';
+    }
+
+    return null;
+  }
+
+  assertFound(variableString, valueToPopulate) {
```

Comment:
Why do we need two methods? Can't we integrate this handling in existing `warnIfNotFound` (which we can rename into e.g. `handleUnresolved`)

Reply:
I wasn't sure if the current method was being used by any plugins, and so wanted to keep the breaking change as small as possible.

Happy to do it as one method if you're fine with that - I went with `assertFound` for the name as that follows the name for methods in the `assert` module and so indicates that the method is an [assertion guard](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions).

Reply:
> I wasn't sure if the current method was being used by any plugins

I think it's very unlikely, also it's considered as _private_ part, so plugins should really not do that

Reply:
> Do you want me to do any form of underscore naming to highlight their private? I've noticed that's done in a couple of other places in the framework :)

I believe it's only done in places where we want to distinct _private_ API from _public_. Here it's not the case (whole class is _private_)

Reply:
Fair enough - want me to stick an `@private` on the class' jsdoc block?